qiskit_nature/mappers/second_quantization/bksf.py

```diff
@@ -0,0 +1,521 @@
+    """Calculate the edge operator A_ij.
+
+    The definitions used here are consistent with arXiv:quant-ph/0003137
+
+    Args:
+        edge_list (numpy.ndarray): a 2xE matrix, where E is total number of edge
+                                    and each pair denotes (from, to)
+        i (int): specifying the edge operator A
+        j (int): specifying the edge operator A
+
+    Returns:
+        Pauli: qubit operator
+    """
+    v = np.zeros(edge_list.shape[1])
+    w = np.zeros(edge_list.shape[1])
+
+    position_ij = -1
+    qubit_position_i = np.asarray(np.where(edge_list == i))
+
+    for edge_index in range(edge_list.shape[1]):
+        if set((i, j)) == set(edge_list[:, edge_index]):
+            position_ij = edge_index
+            break
+
+    w[position_ij] = 1
+
+    for edge_index in range(qubit_position_i.shape[1]):
+        i_i, j_j = qubit_position_i[:, edge_index]
+        i_i = 1 if i_i == 0 else 0  # int(not(i_i))
+        if edge_list[i_i][j_j] < j:
+            v[j_j] = 1
+
+    qubit_position_j = np.asarray(np.where(edge_list == j))
+    for edge_index in range(qubit_position_j.shape[1]):
+        i_i, j_j = qubit_position_j[:, edge_index]
+        i_i = 1 if i_i == 0 else 0  # int(not(i_i))
+        if edge_list[i_i][j_j] < i:
+            v[j_j] = 1
+
+    qubit_op = Pauli((v, w))
+    return SparsePauliOp(qubit_op)
+
+
+def edge_operator_bi(edge_list: np.ndarray, i: int) -> SparsePauliOp:
+    """Calculate the edge operator B_i.
+
+    The definitions used here are consistent with arXiv:quant-ph/0003137
+
+    Args:
+        edge_list (numpy.ndarray): a 2xE matrix, where E is total number of edge
+                                    and each pair denotes (from, to)
+        i (int): index for specifying the edge operator B.
+
+    Returns:
+        Pauli: qubit operator
+    """
+    qubit_position_matrix = np.asarray(np.where(edge_list == i))
+    qubit_position = qubit_position_matrix[1]
+    v = np.zeros(edge_list.shape[1])
+    w = np.copy(v)
+    v[qubit_position] = 1
+    qubit_op = Pauli((v, w))
+    return SparsePauliOp(qubit_op)
+
+
+def _add_sparse_pauli(qubit_op1: SparsePauliOp, qubit_op2: SparsePauliOp) -> SparsePauliOp:
+    """
+    Return `qubit_op1` and `qubit_op2`, except when either one is `None`.
+    In the latter case, return the one that is not `None`. In other words, assume
+    `None` signifies the additive identity.
+    """
+    if qubit_op1 is None:
+        return qubit_op2
+    elif qubit_op2 is None:
+        return qubit_op1
+    else:
+        return qubit_op1 + qubit_op2
+
+
+def _to_physicist_index_order(facs: List) -> Tuple[List, int]:
+    """
+    Reorder the factors `facs` to be two raising operators followed by two lowering operators and
+    return the new factors and the phase incurred by the reordering. Note that `facs` are not in
+    chemists' order, but rather sorted by index with least index first.
```

Comment:
This is normal ordering, correct?
@ikkoham What's the status on #88?
@jlapeyre Is that something you could make use of instead? Then we don't need to duplicate this logic.

Reply:
The relation between #88 and the present ordering function is not clear to me.

Reply:
@ikkoham could you please take a look at this? Maybe this is clearer to you as the author of #88

Reply:
What happened to this discussion?

Reply:
This indeed is the normal ordering of a fermionic operator. We should address this.

Reply:
Ok. I'll take another look into it.

Reply:
Sorry for late. I did not notice the mention. I have updated #88. Previously, special class was needed to represent the normal order, but now we have the sparse label, I improved the implementation using the sparse label.

Reply:
I tested #88. This indeed looks similar to what I have done here. However, the sparse representation of `FermionicOp` is a space separated string. I would probably need to decompose it into convenient data structures, lists and tuples, as I have done here.

Apparently, there are no methods in the class to parse the string into data structures. Is this correct?

Reply:
tagging also @ikkoham here. I think #88 is ready to be merged. So let's clarify this open point so that we finalize this also. Personally I think we can add a TODO as a reminder here and possibly address it in the future, but maybe it is worthy to do now.